security/pkg/server/ca/server.go

```diff
@@ -195,11 +197,16 @@ func New(
 	if len(features.CATrustedNodeAccounts) > 0 && client != nil {
 		// TODO: do we need some way to delayed readiness until this is synced? Probably
 		// Worst case is we deny some requests though which are retried
-		na, err := NewNodeAuthorizer(client, filter, features.CATrustedNodeAccounts)
+		na, err := NewClusterNodeAuthorizer(client, filter, features.CATrustedNodeAccounts)
 		if err != nil {
 			return nil, err
 		}
-		server.nodeAuthorizer = na
+		mNa := NewMulticlusterNodeAuthenticator(filter, features.CATrustedNodeAccounts, addClusterHandler)
```

Comment:
Your points are well taken. Based on our understanding, the race issue arises because we currently cannot guarantee that the `MultiClusterNodeAuthorizer` is added to the handlers before `Run()`. Since the Istiod CA Server is initialized during the `discoveryServer.Start` phase, it is difficult to solve this issue solely by modifying the node authorizer code. In our latest commit, we have modified the CA Server startup process; the CA Server is now initialized first during the `bootstrap.NewServer` phase and then registered with the grpc server during the `discoveryServer.Start` phase. This approach ensures that the `MultiClusterNodeAuthorizer` is added to the handlers before `Run()`, as this action is completed during the CA Server's initialization process. Additionally, with this modification, we no longer need to manually create and add the primary cluster's `ClusterNodeAuthorizer`; it can be created and added through `c.handleAdd(configCluster, stopCh)`. Please review this and thank you for your input.

Reply:
Thanks for pointing this out. Based on our understanding, the race issue arises because we currently cannot guarantee that the `MultiClusterNodeAuthorizer` is added to the handlers before `Run()`. Since the Istiod CA Server is initialized during the `discoveryServer.Start` phase, it is difficult to solve this issue solely by modifying the node authorizer code. In our latest commit, we have modified the CA Server startup process; the CA Server is now initialized first during the `bootstrap.NewServer` phase and then registered with the grpc server during the `discoveryServer.Start` phase. This approach ensures that the `MultiClusterNodeAuthorizer` is added to the handlers before `Run()`, as this action is completed during the CA Server's initialization process. Additionally, with this modification, we no longer need to manually create and add the primary cluster's `ClusterNodeAuthorizer`; it can be created and added through `c.handleAdd(configCluster, stopCh)`. Please review this and thank you again for your input.